rocAL/rocAL/source/augmentations/geometry_augmentations/node_resize_single_param.cpp

```diff
@@ -0,0 +1,102 @@
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include <vx_ext_rpp.h>
+#include <graph.h>
+#include "node_resize_single_param.h"
+#include "exception.h"
+
+
+ResizeSingleParamNode::ResizeSingleParamNode(const std::vector<Image *> &inputs, const std::vector<Image *> &outputs) :
+        Node(inputs, outputs)
+{
+}
+
+void ResizeSingleParamNode::create_node()
+{
+    if(_node)
+        return;
+
+    _dest_width_val.resize(_batch_size);
+    _dest_height_val.resize(_batch_size);
+
+    std::vector<uint32_t> dst_roi_width(_batch_size,_outputs[0]->info().width());
+    std::vector<uint32_t> dst_roi_height(_batch_size, _outputs[0]->info().height_single());
+
+    _dst_roi_width = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, _batch_size);
+    _dst_roi_height = vxCreateArray(vxGetContext((vx_reference)_graph->get()), VX_TYPE_UINT32, _batch_size);
+
+    vx_status width_status, height_status;
+
+    width_status = vxAddArrayItems(_dst_roi_width, _batch_size, dst_roi_width.data(), sizeof(vx_uint32));
+    height_status = vxAddArrayItems(_dst_roi_height, _batch_size, dst_roi_height.data(), sizeof(vx_uint32));
+     if(width_status != 0 || height_status != 0)
+        THROW(" vxAddArrayItems failed in the resize (vxExtrppNode_Resizetensor) node: "+ TOSTR(width_status) + "  "+ TOSTR(height_status))
+
+   _node = vxExtrppNode_Resizetensor(_graph->get(), _inputs[0]->handle(), _src_roi_width, _src_roi_height, _outputs[0]->handle(), _dst_roi_width, _dst_roi_height, _batch_size);
+
+    vx_status status;
+    if((status = vxGetStatus((vx_reference)_node)) != VX_SUCCESS)
+        THROW("Adding the resize (vxExtrppNode_Resizetensor) node failed: "+ TOSTR(status))
+
+}
+
+
+
+void ResizeSingleParamNode::update_node()
+{
+
+std::vector<uint32_t> src_roi_width, src_roi_height;
+src_roi_width = _inputs[0]->info().get_roi_width_vec();
+src_roi_height = _inputs[0]->info().get_roi_height_vec();
+
+for(uint i = 0; i < _batch_size; i++)
+    {
+
+        uint w = src_roi_width[i];
+        uint h = src_roi_height[i];
+
+        uint old_short, old_long, new_short, new_long, new_w, new_h;
+
+        old_short = (w <= h) ? w : h;
+        old_long = (w <= h) ? h : w;
+        new_short = _size;
+        new_long = int(_size * old_long / old_short);
```

Comment:
use static_cast here

Also new_long is a uint type and you have type casted to int here please check